src/resolve/resolved-dns-query.c

```diff
@@ -584,6 +585,15 @@ void dns_query_complete(DnsQuery *q, DnsTransactionState state) {
 
         q->state = state;
 
+        if (set_size(q->manager->varlink_subscription) > 0) {
+                DnsQuestion* question = q->request_packet ? q->request_packet->question :
+                                                            NULL;
+                const char* query_name = question ? dns_question_first_name(question) : q->request_name;
+                int r = send_dns_notification(q->manager, q->answer, query_name);
+                if (r < 0)
+                        log_error_errno(r, "Failed to send varlink notification: %m");
```

Comment:
this doesn't return, so the log message should be `...notification, ignoring: %m`